impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -89,7 +89,9 @@ private void internalStart() {
         throw new RuntimeException(e);
       }
       try {
-        lock = rw.getChannel().lock();
+        if ((lock = rw.getChannel().tryLock()) == null) {
+          throw new RuntimeException("Couldn't lock rootDir: " + rootDirectory.getAbsolutePath());
```

Comment:
We should probably even be smart about hinting at the difference between inter-process and intra-process locking failures here.